WooCommerce/WooCommerceTests/ViewRelated/Reviews/ReviewsViewControllerTests.swift

```diff
@@ -0,0 +1,101 @@
+import XCTest
+@testable import WooCommerce
+
+/// Tests for `ReviewsViewController`.
+///
+final class ReviewsViewControllerTests: XCTestCase {
+    private var mockViewModel: MockReviewsViewModel!
+    private var sut: ReviewsViewController!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        mockViewModel = MockReviewsViewModel(siteID: 123)
+        sut = ReviewsViewController(viewModel: mockViewModel)
+    }
+
+    override func tearDownWithError() throws {
+        mockViewModel = nil
+        sut = nil
+
+        try super.tearDownWithError()
+    }
+
+    func test_mark_all_as_ready_bar_button_is_not_present_if_there_are_no_unread_notifications() {
+        // When
+        mockViewModel.hasUnreadNotifications = false
+        sut.makeViewAppear()
+
+        // Then
+        XCTAssertNil(sut.navigationItem.rightBarButtonItem)
+    }
+
+    func test_mark_all_as_ready_bar_button_is_visible_if_there_are_unread_notifications_available() throws {
+        // When
+        mockViewModel.hasUnreadNotifications = true
+        sut.makeViewAppear()
+
+        // Then
+        let markAllAsReadyButton = try XCTUnwrap(sut.navigationItem.rightBarButtonItem)
+        XCTAssertEqual(markAllAsReadyButton.accessibilityIdentifier, "reviews-mark-all-as-read-button")
+    }
+}
+
+// MARK: ReviewsViewController helpers
+//
+private extension ReviewsViewController {
+    func makeViewAppear() {
+        loadViewIfNeeded()
+        beginAppearanceTransition(true, animated: false)
+        endAppearanceTransition()
+    }
```

Comment:
I'm curious: can you please explain the reason why these methods need to be called when testing the view controller?

Reply:
These methods simulate the act of presenting a view controller. (The view controller's viewDidLoad and viewWillAppear methods are called.)